x/erc20/keeper/token_pairs.go

```diff
@@ -143,10 +145,50 @@ func (k Keeper) IsDenomRegistered(ctx sdk.Context, denom string) bool {
 	return store.Has([]byte(denom))
 }
 
+// GetCoinAddress returns the corresponding ERC-20 contract address for the
+// given denom.
+// If the denom is not registered and its an IBC voucher, it returns the address
+// from the hash of the ICS20's DenomTrace Path.
+func (k Keeper) GetCoinAddress(ctx sdk.Context, denom string) (contractAddress common.Address, ok bool) {
```

Comment:
This function is the analogous of `GetTokenDenom` but they have a different signature: one is returning an error and the other one a bool. I would stay consistent and choose only one of the two.